compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -760,7 +760,11 @@ fn transform_predicates<'tcx>(
             ty::ExistentialPredicate::AutoTrait(..) => Some(predicate),
         })
         .collect();
-    tcx.mk_poly_existential_predicates(&predicates)
+    if predicates.len() == 0 {
```

Comment:
Above comment is in conflict w/ this hack, for example.

Reply:
Map it to `dyn Drop` or something if you need a dummy principal trait to put in there.